### PR TITLE
Fixes RequiredResources not working in Repeater when DataSource is empty

### LINKED ISSUE #595: Can't add StylesheetResource to Head, it is already rendered
Some of our controls (eg. `Repeater`) are adding controls to `Children` in the `Render` method. When any of the added controls require a new resource it will fail with an exception like *"Can't add StylesheetResource to Head, it is already rendered"*.

We need to always build the control tree before the control is rendered (eg. in `OnLoad` method).

### PR DESCRIPTION
**Pull request by djanosik**
Fixes #595.

### CODE REVIEW

File: src/DotVVM.Framework/Controls/Repeater.cs [hunk 1]
```diff
@@ -97,7 +98,11 @@ public string WrapperTagName
         /// </summary>
         protected internal override void OnLoad(IDotvvmRequestContext context)
         {
-            SetChildren(context);
+            if (context.IsPostBack)
+            {
+                SetChildren(context, true);
```
Comment (djanosik) [inline src/DotVVM.Framework/Controls/Repeater.cs]: @exyi Do we need to populate `Children` in `OnLoad` when the request is not a postback?

Comment (exyi) [reply src/DotVVM.Framework/Controls/Repeater.cs]: Yes we need, we need to build the tree in OnPreRender, it's required for `Postback.Update` to work and we need to render it in OnLoad, it's needed for invocation of a command in the children.

Comment (djanosik) [reply src/DotVVM.Framework/Controls/Repeater.cs]: Exactly. But it is enabled for postbacks. It can't break command invocation, can it?.

Comment (djanosik) [reply src/DotVVM.Framework/Controls/Repeater.cs]: @tomasherceg What do you think about this?

**Pull request merged**